### PR TITLE
Only update file time once

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes issue with continous testing intermittent failures

### CODE REVIEW

File: test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java [hunk 1]
```diff
@@ -539,11 +539,13 @@ public void modifyTestSourceFile(Class<?> sourceFile, Function<String, String> m
      * @param sourceFile
      */
     public void addSourceFile(Class<?> sourceFile) {
-        Path path = copySourceFilesForClass(projectSourceRoot, deploymentSourcePath, testLocation,
+        Path path = findTargetSourceFilesForPath(projectSourceRoot, deploymentSourcePath, testLocation,
+                testLocation.resolve(sourceFile.getName().replace(".", "/") + ".class"));
+        long old = modTime(path.getParent());
```
Comment (gsmet) [inline test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java:544]: `findTargetSourceFilesForPath` can return `null` and you don't test it AFAICS. Maybe it's a corner case but it seems to be an open door for NPEs.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by gsmet** [comment]
@stuartwdouglas I'm not sure it's related to your work but this `DevServicesMySQLDatasourceTestCase.testDatasource` started failing very recently. It could be a GH Action infra issue though.

File: test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java (hunk 1 above)
Comment (stuartwdouglas) [reply test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java:544]: The original copySourceFilesForClass could also return null. I will make them throw an exception instead.

**Comment by stuartwdouglas** [comment]
I think DevServicesMySQLDatasourceTestCase.testDatasource is failing due to a JDK upgrade on the GH actions infrastructure. Should be fixed by https://github.com/quarkusio/quarkus/pull/16746

**Comment by stuartwdouglas** [comment]
@gsmet it would be good to get this in, it will hopefully fix the continuous testing intermittent failures.

**Review by gsmet — APPROVED** [review]

**Pull request merged**